### DISCUSSION TITLE
Generic types fo WhereInputs (and co)

### DISCUSSION THREAD

**Discussion by tyteen4a03** [prisma/prisma#14310] [Q&A]
I'm currently working on writing a generic pagination querystring to query generator for Prisma. However, it seems like the generated types into `@prisma/client` doesn't contain generic versions of `ModelWhereInput` so it's very hard for me to write generic code that ties into all `WhereInput`s.

How do you approach this issue? Or are there existing solutions / PRs?

**Comment by nurul3101** [comment]
Hey @tyteen4a03 👋 

Generated types do contain generic `ModelWhereInput` types.
For example, I have a model named Person and this is the `PersonWhereInput` type which gets generated.

```typescript
  export type PersonWhereInput = {
    AND?: Enumerable<PersonWhereInput>
    OR?: Enumerable<PersonWhereInput>
    NOT?: Enumerable<PersonWhereInput>
    id?: IntFilter | number
    first_name?: StringFilter | string
    last_name?: StringFilter | string
    team_id?: IntNullableFilter | number | null
    personData?: XOR<PersonDataRelationFilter, PersonDataWhereInput> | null
    team?: XOR<TeamRelationFilter, TeamWhereInput> | null
  }
```
Can you share your generated types?

**Reply by tyteen4a03** [reply]
The problem is they are not generic _across_ models, so if I have two models I cannot generalize them into `function <X extends WhereInput>`.

**Reply by nurul3101** [reply]
Yes, they are not generic and are intertwined with the model's fields. Depending on your use case you might want to create a [Feature Request]( https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) to generate Generic types.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
